src/maxdiffusion/models/train.py

```diff
@@ -489,20 +493,18 @@ def compute_loss(unet_params):
         unet_state, train_metric, train_rngs = p_train_step(unet_state,
                                                             example_batch,
                                                             train_rngs)
-        new_time = datetime.datetime.now()
-
-        record_scalar_metrics(train_metric, new_time - last_step_completion, per_device_tflops, learning_rate_scheduler(step))
 
-        step_time_delta = new_time - last_step_completion
-        max_logging.log(f"completed step: {step}, seconds: {step_time_delta.total_seconds()}, "
-          f"TFLOP/s/device: {per_device_tflops / step_time_delta.total_seconds()}, "
-          f"loss: {train_metric['scalar']['learning/loss']:.3f}")
-          
-        if config.write_metrics:
-            write_metrics(writer, local_metrics_file, running_gcs_metrics, train_metric, step, config)
-        last_step_completion = new_time
         step_num = step + 1
         samples_count = total_train_batch_size * step_num
+
+        if config.write_metrics and step % config.metrics_period == 0:
+            new_time = datetime.datetime.now()
+            step_time_delta = new_time - last_step_completion
+            record_scalar_metrics(train_metric, step_time_delta, config.metrics_period, per_device_tflops, learning_rate_scheduler(step))
+            write_metrics(writer, local_metrics_file, running_gcs_metrics, train_metric, step, config)
+            mllog_utils.maybe_train_step_log(config, start_step, step_num, samples_count, train_metric)
```

Comment:
we can align `train_log_interval` with config. metrics_period

Reply:
Thanks for catching this!